Simplenote/PasskeyAuthenticator.swift

```diff
@@ -3,126 +3,105 @@ import AuthenticationServices
-        authController.delegate = self
-        authController.presentationContextProvider = presentationContext
-        authController.performRequests()
+        return verifyResponse
     }
+}
 
-    private func performPasskeyRegistration(with credential: ASAuthorizationPlatformPublicKeyCredentialRegistration) {
-        guard let registrationObject = PasskeyRegistrationResponse(from: credential, with: registrationEmail) else {
-            //TODO: Should handle error
-            return
-        }
+typealias PresentationContext = ASAuthorizationControllerPresentationContextProviding
+typealias PublicKeyCredentialAssertion = ASAuthorizationPlatformPublicKeyCredentialAssertion
 
-        Task {
-            do {
-                let data = try JSONEncoder().encode(registrationObject)
-                try await passkeyRemote.registerCredential(with: data)
-            } catch {
-                //TODO: Display error
-            }
-        }
+class PasskeyAuthenticator: NSObject {
+    let passkeyRemote: PasskeyRemote
+    let internalAuthControllerDelegate: PasskeyAuthControllerDelegate
+
+    init(passkeyRemote: PasskeyRemote = PasskeyRemote(), authControllerDelegate: PasskeyAuthControllerDelegate = .init()) {
+        self.passkeyRemote = passkeyRemote
+        self.internalAuthControllerDelegate = authControllerDelegate
     }
 
-    // MARK: - Auth
-    //
-    func attemptPasskeyAuth(for email: String, in presentationContext: PresentationContext) async throws {
-        guard let challenge = try await fetchAuthChallenge(for: email) else {
-            return
+    func attemptPasskeyAuth(challenge: PasskeyAuthChallenge?, in presentationContext: PresentationContext) async throws -> PasskeyVerifyResponse {
+        guard let challenge else {
+            throw PasskeyError.couldNotFetchAuthChallenge
         }
 
         let challengeData = try Data.decodeUrlSafeBase64(challenge.challenge)
         let provider = ASAuthorizationPlatformPublicKeyCredentialProvider(relyingPartyIdentifier: challenge.relayingParty)
         let request = provider.createCredentialAssertionRequest(challenge: challengeData)
 
         let controller = ASAuthorizationController(authorizationRequests: [request])
-        controller.delegate = self
+        controller.delegate = internalAuthControllerDelegate
         controller.presentationContextProvider = presentationContext
-        controller.performRequests()
+
+        return try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<PasskeyVerifyResponse, any Error>) in
+            internalAuthControllerDelegate.onCompletion = { result in
+                switch result {
+                case .success(let verify):
+                    continuation.resume(returning: verify)
+                case .failure(let error):
+                    continuation.resume(throwing: error)
+                }
+            }
+
+            controller.performRequests()
```

Comment:
Is this API okay with BG invocation? (we're not necessarily in the main Task)

Reply:
should be I believe.  The finishing and ui changes happen on the main thread, the auth can happen in the background I think